### ISSUE TITLE
TASK [installing ascs] failed for S4HANA2021 on NetWeaver-HA

### ISSUE THREAD

**Issue by danmk3** [GoogleCloudPlatform/sap-deployment-automation#48]
I am installing S4HANA 2021 on rhel-8-1-sap-ha and when it reaches this step it failes. I have HANA  2.00.057.00.1629894416 so the HANA in HA is installed OK and it is in HSR.
Full log is attached and first part of the error : 
```

TASK [ascs : inifile.params config file] ****************************************************************************************************************************************
changed: [172.30.1.219]

TASK [installing ascs] **********************************************************************************************************************************************************
fatal: [172.30.1.219]: FAILED! => {"changed": true, "cmd": ["./sapinst", "SAPINST_USE_HOSTNAME=ascsvirt", "SAPINST_INPUT_PARAMETERS_URL=/usr/sap/inifile_ascs.params", "SAPINST_EXECUTE_PRODUCT_ID=NW_ABAP_ASCS:S4HANA2021.CORE.HDB.ABAPHA", "SAPINST_SKIP_DIALOGS=true", "SAPINST_START_GUISERVER=false"], "delta": "0:00:22.036791", "end": "2021-12-06 12:26:13.371622", "msg": "non-zero return code", "rc": 111, "start": "2021-12-06 12:25:51.334831", "stderr": "=>sapparam(1c): No Profile used.\n=>sapparam: SAPSYSTEMNAME neither in Profile nor in Commandline\n################################################\nAbort execution because of \nStep returns modlib.jslib.caughtException\n################################################", "stderr_lines": ["=>sapparam(1c): No Profile used.", "=>sapparam: SAPSYSTEMNAME neither in Profile nor in Commandline", "################################################", "Abort execution because of ", "Step returns modlib.jslib.caughtException", "################################################"], "stdout": "Extracting...\nExtraction done!\nINFO: Native locale set to C.UTF-8\nINFO       2021-12-06 12:25:53.365 (mainThread) [sixxcreate.cpp:349]\n********************************************************************************\nInitial log directory: /root/.sapinst/sascs-1/31590\n********************************************************************************\n\n\nSAPinst build information:\n--------------------------\nVersion:         749.0.92\nBuild:           2091532\nCompile time:    Sep 28 2021 - 15:17:06\nMake type:       optU\nCodeline:        749_REL\nPlatform:        linuxx86_64\nKernel build:    749, patch 1026, changelist 2091140\nSAP JRE build:   SAP Java Server VM (build 8.1.080 11.0.12+000, Sep 14 2021 16:18:20 - 81_REL - optU - linux amd64 - 6 - bas2:330008 (mixed mode))\nSAP JCo build:   3.0.21\nSL-UI version:   2.6.67\nSAP UI5 version: 1.60.30\nExe directory: /tmp/sapinst_exe.31587.1638793551\n\nSAPinst process information:\n----------------------------\nPid:        31590\n\nload resource pool /sapmnt/Software/SWPM2.0/resourcepool.xml\nWARNING    2021-12-06 12:25:53.368 (root/sapinst) (startInstallation) [syxxcnwmgt.cpp:61] id=syslib.network.overriddenHostName void CSyNetworkMgtImpl::overrideHostName(const iastring & \"ascsvirt\")\nHost name has been overridden. Using host name 'ascsvirt' which might not be the real host name.\n\nINFO       2021-12-06 12:25:53.517 (root/sapinst) (startInstallation) [syuxcfile.cpp:407] id=syslib.filesystem.copiedFile CSyFileImpl::copy(const CSyPath & /root/.sapinst/sascs-1/31590, ISyNode::CopyMoveMode_t 0x3, ISyProgressObserver*) const \nCopied file '/sapmnt/Software/SWPM2.0/dev_sap_kernel' to '/root/.sapinst/sascs-1/31590'.\n\nINFO       2021-12-06 12:25:53.521 (root/sapinst) (startInstallation) [syuxcfile.cpp:216] id=syslib.filesystem.removedFile CSyFileImpl::removeEx(ISyFSErrorHandler * pErrorHandler)\nRemoved file /sapmnt/Software/SWPM2.0/dev_sap_kernel.\n\nINFO       2021-12-06 12:25:53.523 (root/sapinst) (startInstallation) [CSiManagerInterfaces.cpp:4011] CSiManagerInterfaces::setSWPMSupportPackageFromManifestFile()\nRunning Software Provisioning Manager Key Release 2.0 SP 10\n\nINFO       2021-12-06 12:25:53.701 (root/sapinst) (startInstallation) [syuxcaccmg.cpp:254] id=syslib.account.isExistingOnOsSucceededFalse bool CSyAccountMgtImpl::isExisting(ISyAccountMgt::eAccountType accountType, iastring sName)\nAccount sapinst does not exist.\n\nINFO[E]    2021-12-06 12:25:53.703 (root/sapinst) (startInstallation) [syuxcgroup.cpp:976] id=common.functionReturn errno=FSH-00006 CSyGroupImpl_getOsInfos(sapinst, )\nReturn value of function getgrnam(sapinst) is Success.\n\nINFO       2021-12-06 12:25:53.967 (root/sapinst) (startInstallation) [syuxcgroup.cpp:895] id=syslib.account.creationSucceeded CSyGroupImpl::addToOS(ISyAccountMgt::eAccountType, ISyProgressObserver*)\nAccount group=\"sapinst\" gid=\"1004\" created.\n\nINFO  
```

[sap-deployment-automation on gcp rhel 81 - error installing ascs.txt](https://github.com/GoogleCloudPlatform/sap-deployment-automation/files/7660756/sap-deployment-automation.on.gcp.rhel.81.-.error.installing.ascs.txt)
.

**Comment by rjosephwright** [comment]
Hi @danmk3, I have not tested S4HANA 2021, but in 2020 the password needs to be 10-14 characters. Based on the error below, I think that the value for `sap_nw_password` is too short.

```
ERROR      2021-12-06 12:26:10.821 (root/sapinst) (startInstallation) [iaxxbjsmod.cpp:83] id=modlib.jslib.caughtException errno=MUT-03025
Caught ESAPinstException in module call: Validator of step '|NW_ABAP_ASCS|ind|ind|ind|ind|0|0|NW_GetSidNoProfiles|ind|ind|ind|ind|getSid|0|NW_GetUserParameter|ind|ind|ind|ind|users|0|NW_GetMasterPassword|ind|ind|ind|ind|masterpwd|0|askMasterPwd' reported an error:
Start SAPinst in interactive mode to solve this problem.
```

**Comment by danmk3** [comment]
I have passed that error as this time I tried with a password that contained 10 characters. It managed to install ascs and ers. Now I have another error on the pas server at step:
TASK [installing db]. It seems unzip the S4HANA2021 archive export .

`
\nCopied file '/tmp/sapinst_instdir/S4HANA2021/CORE/HDB/INSTALL/HA/ABAP/DB/ArchiveScanResults.html' to '/tmp/sapinst_instdir/S4HANA2021/CORE/HDB/INSTALL/HA/ABAP/DB/ArchiveScanResults.3.html'.\n\nINFO       2021-12-06 17:35:21.475 (root/sapinst) (startInstallation) [ArchiveCheckerInformationWriterLogfile.cpp:36]\nArchive scan summary for '/sapmnt/Software/Kernel_Files'; see also /tmp/sapinst_instdir/S4HANA2021/CORE/HDB/INSTALL/HA/ABAP/DB/ArchiveScanResults.3.html\n********************************\nRequested Archives: BACKUP1 (checks are listed in reverse order)\nChecking archive 'S4HANA2021CORE HANA DB Export 1' ('BACKUP1'): Not found\n   Manifest File\n      -- '/sapmnt/Software/Kernel_Files/igsexe_0-80001746.sar': Archive does not contain manifest file 'META-INF/SAPMANIFEST.MF'. Found manifest file(s): igsmanifest.mf.\n      -- '/sapmnt/Software/Kernel_Files/igshelper_4-10010245.sar': Archive does not contain manifest file 'META-INF/SAPMANIFEST.MF'. Found manifest file(s): igs/igshelpermanifest.mf.\n      -- '/sapmnt/Software/Kernel_Files/SAPHOSTAGENT53_53-80004822.SAR': Archive does not contain manifest file 'META-INF/SAPMANIFEST.MF'. Found manifest file(s): hostagent.mf.\n      -- '/sapmnt/Software/Kernel_Files/SAPEXE_50-80005374.SAR': Archive does not contain manifest file 'META-INF/SAPMANIFEST.MF'. Found manifest file(s): sapmanifest.mf, webdispatchermanifest.mf, sapcrypto.mf.\n      -- '/sapmnt/Software/Kernel_Files/SAPEXEDB_50-80005373.SAR': Archive does not contain manifest file 'META-INF/SAPMANIFEST.MF'. Found manifest file(s): sapmanifestdb.mf.\n\n********************************\nAutomatically found Archives: -\n\nINFO       2021-12-06 17:35:21.505 (root/sapinst) (startInstallation) [CSiStepExecute.cpp:971] id=controller.executeStep CSiStepExecute::execute()\nExecute step SpecifyArchive of component |NW_ABAP_DB|ind|ind|ind|ind|0|0|NW_CreateDBandLoad|ind|ind|ind|ind|createdbandload|0|nw_get_installation_backup|ind|ind|ind|ind|get_installation_backup|0|nw_get_installation_backup_archives|ind|ind|ind|ind|archives|0|archive_extraction|ind|ind|ind|ind|first|0\n\nWARNING    2021-12-06 17:35:21.580 (root/sapinst) (startInstallation) id=cdsrv.archiveCheckerStatus.furtherArchivesRequired com.sap.sdt.ins.component.archive_extraction.SpecifyArchive\nThe following archives are also required, otherwise you cannot continue: \"S4HANA2021CORE HANA DB Export 1\"\n\nERROR      2021-12-06 17:35:21.580 (root/sapinst) (startInstallation) id=cdsrv.archiveCheckerStatus.downloadbasketNotSet com.sap.sdt.ins.component.archive_extraction.SpecifyArchive\nParameter for download basket is empty.\n\nERROR      2021-12-06 17:35:21.652 (root/sapinst) (startInstallation) [CSiStepExecute.cpp:1104] id=controller.stepExecuted errno=FCO-00011 CSiStepExecute::execute()\nThe step SpecifyArchive with step key 
`
Full log here:
[installing db pas error.txt](https://github.com/GoogleCloudPlatform/sap-deployment-automation/files/7662585/installing.db.pas.error.txt)

**Comment by rjosephwright** [comment]
As I mentioned previously, this version has not been tested by me, so I cannot tell you exactly what files you need to install this. However, it looks to me like you are missing DB export files. The message `Parameter for download basket is empty` usually means there is a missing file, and in this case it also says `The following archives are also required, otherwise you cannot continue: \"S4HANA2021CORE HANA DB Export 1\"`.

**Comment by danmk3** [comment]
I  managed to install it by placing the export kits and extracted media next to all media ( near kernel files, etc). In the end I was able to login to the system and the HANA HSR was working. After the DB instance was done almost at the end of the ansible script I got a new error : 

```
TASK [sap-assertions : fail] 
fatal: [172.30.1.2]: FAILED! => {"changed": false, "msg": "number of tries exceeded waiting for pacemaker"}
fatal: [172.30.1.4]: FAILED! => {"changed": false, "msg": "number of tries exceeded waiting for pacemaker"}
```
There is a problem with the stonith:fence_gce that I have but this is another issue with my google IAM roles and permissions.
"Required 'compute.instances.list' permission for 'projects/development-project-xxxxxxx'">
